lib/active_storage/service/open_stack_service.rb

```diff
@@ -16,10 +16,9 @@ def initialize(container:, credentials:, connection_options: {})
 
     def upload(key, io, checksum: nil)
       instrument :upload, key: key, checksum: checksum do
-        params = {
-          'Content-Type' => guess_content_type(io),
-          'ETag' => convert_base64digest_to_hexdigest(checksum)
-        }
+        params = { 'Content-Type' => guess_content_type(io) }
+        params['ETag'] = convert_base64digest_to_hexdigest(checksum) if checksum
```

Comment:
The purpose of this gem is to be compatible with all OpenStack providers. I'm going to approve and merge this PR, but could you **send me** the OpenStack wrapper library you're using. I might submit a PR to rails as to send the checksum with variants as well ( I don't get why they don't as you've mentioned)

Reply:
Do you mean the OpenStack wrapper library from my cloud provider?

Reply:
Yes